sql/src/main/java/io/crate/execution/dsl/projection/WindowAggProjection.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.dsl.projection;
+
+import com.google.common.collect.ImmutableMap;
+import io.crate.analyze.WindowDefinition;
+import io.crate.execution.dsl.projection.builder.InputColumns;
+import io.crate.expression.symbol.Symbol;
+import io.crate.expression.symbol.Symbols;
+import io.crate.expression.symbol.WindowFunction;
+import io.crate.planner.ExplainLeaf;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class WindowAggProjection extends Projection {
+
+    private final List<WindowFunction> windowFunctions;
+    private final Map<WindowDefinition, List<Symbol>> functionsByWindow;
+    private final List<Symbol> inputs;
+
+    public WindowAggProjection(List<Symbol> inputs, List<WindowFunction> windowFunctions) {
```

Comment:
In the other projections we follow the pattern that the projection already receives the symbols with inputColumns. I think we should follow that here as well. It removes the need to hold onto the inputs which then don't have to be streamed. It would also avoid doing the grouping multiple times (on the planner + every node on deserialization - if it's being streamed) (Unless I'm missing something?)